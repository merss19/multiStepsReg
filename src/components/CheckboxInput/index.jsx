"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_router_dom_1 = require("react-router-dom");
var CheckboxInput = /** @class */ (function (_super) {
    __extends(CheckboxInput, _super);
    function CheckboxInput() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CheckboxInput.prototype.render = function () {
        var _a = this.props, input = _a.input, title = _a.title, id = _a.id, _b = _a.meta, touched = _b.touched, error = _b.error;
        return (<div className='checkboxes__item'>
				<span className='checkbox'>
					<label className='checkbox__label' htmlFor={id}>
						<input {...input} className='checkbox__field' checked={input.value} id={id} type='checkbox'/>
						<span className='checkbox__ph'>
						<svg className='svg ico-tick'>
							<use xlinkHref='#ico-tick'/>
						</svg>
				 </span>
					<div className='checkbox__group'>
						<span className='checkbox__title'>{title}</span>
							<react_router_dom_1.Link target='_blank' className='checkbox__link' to='http://todayme.ru/dogovor-oferty#.WGFQqrZ95E4'>
								оферты
							</react_router_dom_1.Link>
							<span className='checkbox__divider'>и</span>
							<react_router_dom_1.Link target='_blank' className='checkbox__link' to='https://todayme.ru/rules.pdf'>
								правил
							</react_router_dom_1.Link>
					</div>
				</label>
				</span>
        {touched && error && <span className='checkbox__error'>{error}</span>}
      </div>);
    };
    return CheckboxInput;
}(React.Component));
exports.default = CheckboxInput;
