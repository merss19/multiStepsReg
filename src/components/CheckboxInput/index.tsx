import * as React from 'react';
import {WrappedFieldProps} from 'redux-form';
import {Link} from 'react-router-dom';

interface OwnProps extends WrappedFieldProps<{}> {
  id: string
  title: string
}

class CheckboxInput extends React.Component<OwnProps, {}> {
  render() {
    const {input, title, id, meta: {touched, error}} = this.props;
    return (
      <div className='checkboxes__item'>
				<span className='checkbox'>
					<label className='checkbox__label' htmlFor={id}>
						<input
        {...input}
        className='checkbox__field'
        checked={input.value}
        id={id}
        type='checkbox'
      />
						<span className='checkbox__ph'>
						<svg className='svg ico-tick'>
							<use xlinkHref='#ico-tick'/>
						</svg>
				 </span>
					<div className='checkbox__group'>
						<span className='checkbox__title'>{title}</span>
							<Link
         target='_blank'
         className='checkbox__link'
         to='http://todayme.ru/dogovor-oferty#.WGFQqrZ95E4'
       >
								оферты
							</Link>
							<span className='checkbox__divider'>и</span>
							<Link
         target='_blank'
         className='checkbox__link'
         to='https://todayme.ru/rules.pdf'
       >
								правил
							</Link>
					</div>
				</label>
				</span>
        {touched && error && <span className='checkbox__error'>{error}</span>}
      </div>

    )
  }
}

export default CheckboxInput;