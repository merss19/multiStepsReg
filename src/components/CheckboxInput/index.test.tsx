import {shallow} from 'enzyme';
import CheckboxInput from './index';
import * as React from 'react';
import toJson from 'enzyme-to-json';

import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

let props = {
  id: 'offert',
  title: 'Оферта',
  meta: {
    touched: true,
    error: 'Email должен быть заполнен',
    asyncValidating: false,
    autofilled: false,
    invalid: false,
    pristine: false,
    dirty: false,
    dispatch: (fn: any) => fn,
    valid: false
  },
};
let id;
let title;
let touched;
let value;
const start = () => {
  beforeEach(() => {
    id = 'offert',
    title = 'Оферта',
    touched = false,
    value = ''
  });
  const props = {
    id: id,
    title: title,
    input: {
      name: 'accept',
      onBlur:jest.fn(),
      onChange:jest.fn(),
      onDragStart:jest.fn(),
      onDrop:jest.fn(),
      onFocus:jest.fn(),
      value: value
    },
    meta: {
      touched: touched,
      error: 'Email должен быть заполнен',
      asyncValidating: false,
      autofilled: false,
      invalid: false,
      pristine: false,
      dirty: false,
      dispatch: (fn: any) => fn,
      valid: false
    }
  };
  const buildComponent = () => {
    return shallow(<CheckboxInput {...props} />)
  };
  return {
    props,
    buildComponent
  }
};
describe('Component CheckboxInput', () => {
  it('renders component', () => {
    const component = start().buildComponent();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(start().buildComponent());
    expect(tree).toMatchSnapshot();
  });
  it('renders title', () => {
    const component = start().buildComponent();
    expect(component.find('.checkbox__title').text()).toBe(start().props.title);
    expect(component.find('.checkbox__error').exists()).toBe(false);
  });
  it('renders id', () => {
    const component = start().buildComponent();
    expect(component.find('.checkbox__label').props().htmlFor).toBe(start().props.id);
    expect(component.find('.checkbox__field').props().id).toBe(start().props.id);
  });
  describe('Render error', () => {
    it('render error', () => {
      touched= true;
      const component = start().buildComponent();
      expect(component.find('.checkbox__error').exists()).toBe(true);
      expect(component.find('.checkbox__error').text()).toEqual(start().props.meta.error);
    });
  });
});