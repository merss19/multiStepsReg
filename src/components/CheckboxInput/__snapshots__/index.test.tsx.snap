// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CheckboxInput renders correctly 1`] = `
<div
  className="checkboxes__item"
>
  <span
    className="checkbox"
  >
    <label
      className="checkbox__label"
      htmlFor="offert"
    >
      <input
        checked=""
        className="checkbox__field"
        id="offert"
        name="accept"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value=""
      />
      <span
        className="checkbox__ph"
      >
        <svg
          className="svg ico-tick"
        >
          <use
            xlinkHref="#ico-tick"
          />
        </svg>
      </span>
      <div
        className="checkbox__group"
      >
        <span
          className="checkbox__title"
        >
          Оферта
        </span>
        <Link
          className="checkbox__link"
          replace={false}
          target="_blank"
          to="http://todayme.ru/dogovor-oferty#.WGFQqrZ95E4"
        >
          оферты
        </Link>
        <span
          className="checkbox__divider"
        >
          и
        </span>
        <Link
          className="checkbox__link"
          replace={false}
          target="_blank"
          to="https://todayme.ru/rules.pdf"
        >
          правил
        </Link>
      </div>
    </label>
  </span>
</div>
`;
